src/wasm/wasm-validator.cpp

```diff
@@ -440,6 +486,17 @@ void FunctionValidator::visitBlock(Block* curr) {
     }
     breakInfos.erase(iter);
   }
+  switch (getFunction()->profile) {
+    case IRProfile::Normal:
+      validateNormalBlockElements(curr);
+      break;
+    case IRProfile::Poppy:
+      validatePoppyBlockElements(curr);
+      break;
+  }
```

Comment:
Are blocks really the only place where we have different validation - that is, in all other cases Poppy validation is just **additional** validation on top of the normal rules? If so it would be good to document that, that's a nice property!

Reply:
Yep! As designed :) I can add a note about that to the docs that are currently in stack-utils.h.